File: 0057-insert-interval/0057-insert-interval.py
class Solution:
  def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
    res = []
    for start, end in sorted(intervals + [newInterval]):
      if res and res[-1][1] >= start:
        res[-1][1] = max(res[-1][1], end)
      else:
        res.append([start, end])

    return res

File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        int n = intervals.size();
        vector<vector<int>> ans;
        int i = 0;
        while (i < n && intervals[i][1] < newInterval[0]) ans.push_back(intervals[i++]);
        while (i < n && intervals[i][0] <= newInterval[1]) newInterval = { min(intervals[i][0], newInterval[0]), max(intervals[i++][1], newInterval[1]) };
        ans.push_back(newInterval);
        while (i < n) ans.push_back(intervals[i++]);
        return ans;
    }
};